examples/version-migration/schema-upgrade/src/modelVersion1/containerCode.ts

```diff
@@ -3,80 +3,82 @@
 
-import type { IInventoryList, IInventoryListAppModel } from "../modelInterfaces.js";
+import type { IInventoryList } from "../modelInterfaces.js";
 
 import { InventoryListAppModel } from "./appModel.js";
 import { InventoryListInstantiationFactory } from "./inventoryList.js";
 
 const inventoryListId = "default-inventory-list";
-const migrationToolId = "migration-tool";
-
-const migrationToolRegistryKey = "migration-tool";
-const migrationToolFactory = new MigrationToolFactory();
 
 /**
  * @internal
  */
-export class InventoryListContainerRuntimeFactory extends ModelContainerRuntimeFactory<IInventoryListAppModel> {
+export class InventoryListContainerRuntimeFactory implements IRuntimeFactory {
+	public get IRuntimeFactory() {
+		return this;
+	}
+
+	private readonly registryEntries = new Map([
+		InventoryListInstantiationFactory.registryEntry,
+	]);
+	private readonly runtimeOptions: IContainerRuntimeOptions | undefined;
 	/**
 	 * Constructor for the factory. Supports a test mode which spawns the summarizer instantly.
 	 * @param testMode - True to enable instant summarizer spawning.
 	 */
 	public constructor(testMode: boolean) {
-		super(
-			new Map([
-				InventoryListInstantiationFactory.registryEntry,
-				[migrationToolRegistryKey, Promise.resolve(migrationToolFactory)],
-			]), // registryEntries
-			testMode
-				? {
-						summaryOptions: {
-							initialSummarizerDelayMs: 0,
-						},
-					}
-				: undefined,
+		this.runtimeOptions = testMode
+			? {
+					summaryOptions: {
+						initialSummarizerDelayMs: 0,
+					},
+				}
+			: undefined;
+	}
+
+	public async instantiateRuntime(
+		context: IContainerContext,
+		existing: boolean,
+	): Promise<IRuntime> {
+		const runtime = await instantiateMigratableRuntime(
+			context,
+			existing,
+			this.registryEntries,
+			this.createModel,
+			this.runtimeOptions,
 		);
+
+		if (!existing) {
+			await this.containerInitializingFirstTime(runtime);
+		}
+
+		return runtime;
 	}
 
-	/**
-	 * {@inheritDoc ModelContainerRuntimeFactory.containerInitializingFirstTime}
-	 */
-	protected async containerInitializingFirstTime(runtime: IContainerRuntime) {
+	private readonly containerInitializingFirstTime = async (runtime: IContainerRuntime) => {
 		const inventoryList = await runtime.createDataStore(
 			InventoryListInstantiationFactory.type,
 		);
 		await inventoryList.trySetAlias(inventoryListId);
-		const migrationTool = await runtime.createDataStore(migrationToolRegistryKey);
-		await migrationTool.trySetAlias(migrationToolId);
-	}
-
-	/**
-	 * {@inheritDoc ModelContainerRuntimeFactory.containerHasInitialized}
-	 */
-	protected async containerHasInitialized(runtime: IContainerRuntime) {
-		// Force the MigrationTool to instantiate in all cases.  The Quorum it uses must be loaded and running in
-		// order to respond with accept ops, and without this call the MigrationTool won't be instantiated on the
-		// summarizer client.
-		await getDataStoreEntryPoint(runtime, migrationToolId);
```

Comment:
This is an example of why it's nice to do on the customer's behalf - this "always instantiate" requirement would be easy to get wrong.  Now the customer doesn't have to think about it.

Note that now this entire container code can remain ignorant to migration except for using the special helper function to instantiate the migratable runtime.